### PR TITLE
blog: Add blog post on Tekton Chains and IBM DevSecOps

### PR DESCRIPTION
**Pull request by lehors**
As discussed on a recent call, Tekton Chains supports SLSA Provenance v1 but the configuration isn't the most straightforward. This post highlights support for SLSA and gives people the right configuration to use to get the v1 format. It also informs people that IBM has an offering based on this technology and gives them a few pointers to the relevant documentation.

### CODE REVIEW

File: docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+The good news is that thanks to a change introduced in [Tekton Chains v0.17.0](https://github.com/tektoncd/chains/releases/tag/v0.17.0), which was released last summer, Tekton Chains can now produce attestations in the SLSA Provenance v1 format. The challenge is that getting the right configuration to enable this capability involves option names that are rather puzzling.
+
+Indeed, one must set the Tekton Chains configuration output format option to `slsa/v2alpha3` to get the Provenance v1 format!?!? While setting the format to `slsa/v1` (which is currently the default) produces an attestation in Provenance v0.2!?!? I know!... but I'm not making this up, it is specified in the [How to configure Tekton Chains](https://tekton.dev/docs/chains/slsa-provenance/#how-to-configure-tekton-chains) documentation. These values were supposed to remain internal and the developers talked about introducing other aliases to be exposed to the user but evidently this hasn't happened yet so `slsa/v2alpha3` is what you must use.
```
Comment (MarkLodato) [inline docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md:23]: I realize this is meant to be tongue-in-cheek, but I worry that it might accidentally be read as disparaging. Perhaps either be a bit softer in tone or get someone from Tekton Chains to see if they agree with that wording?

**Review by MarkLodato — APPROVED** [review]

File: docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md (hunk 1 above)
Comment (lehors) [reply docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md:23]: You're absolutely right. I didn't think about it this way but I don't want to take the risk of offending anyone. I just revised the text to make it more neutral.

**Review by MarkLodato — APPROVED** [review]

File: docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+## From DevOps to DevSecOps
+
+For years the industry has been advocating the modernization of development processes with the adoption of what is known as "DevOps" with the promise that, by combining and automating software development and IT operations, DevOps can speed delivery of higher-quality software. However, with the constant rise in security attacks against all software we've seen a quick shift towards "DevSecOps" which continuously integrates and automates security throughout the DevOps lifecycle. DevSecOps makes security a first class citizen by addressing security concerns throughout the development lifecycle rather than as an afterthought and often last check.
+
+## Tekton Chains
+
+[Tekton](https://tekton.dev) is a powerful and flexible open-source framework for creating continuous integration and delivery (CI/CD) systems which is widely used in the industry to manage the production and deployments of software artifacts in the cloud as part of a DevOps strategy. [Tekton Chains](https://tekton.dev/docs/chains/) adds to Tekton a resource controller which can be used to secure Tekton pipelines and move one to DevSecOps.
```
Comment (arewm) [inline docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md:15]: Is there any documentation that you can link to which discusses how Tekton Chains can help move one to DevSecOps? Or is this just meant to be answered by "with SLSA integration"?

**Review by arewm — APPROVED** [review]

File: docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md (hunk 2 above)
Comment (lehors) [reply docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md:15]: That's pretty much it. Admittedly this statement might be broader than it should but it's at least a step in that direction.

Comment (arewm) [reply docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md:15]: > to secure Tekton pipelines and move one to DevSecOps

Maybe this phrase can be reworded? Chains doesn't secure the pipelines themselves, but it adds transparency to the pipelines via the generated provenance.

Comment (lehors) [reply docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md]: Ok, I changed the sentence to clarify.

**Review by arewm — APPROVED** [review]

File: docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+## From DevOps to DevSecOps
+
+For years the industry has been advocating the modernization of development processes with the adoption of what is known as "DevOps" with the promise that, by combining and automating software development and IT operations, DevOps can speed delivery of higher-quality software. However, with the constant rise in security attacks against all software we've seen a quick shift towards "DevSecOps" which continuously integrates and automates security throughout the DevOps lifecycle. DevSecOps makes security a first class citizen by addressing security concerns throughout the development lifecycle rather than as an afterthought and often last check.
+
+## Tekton Chains
+
+[Tekton](https://tekton.dev) is a powerful and flexible open-source framework for creating continuous integration and delivery (CI/CD) systems which is widely used in the industry to manage the production and deployments of software artifacts in the cloud as part of a DevOps strategy. [Tekton Chains](https://tekton.dev/docs/chains/) adds to Tekton a resource controller which can be used to manage your supply chain security in Tekton pipelines and move you towards DevSecOps.
```
Comment (david-a-wheeler) [inline docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md:15]: I suggest removing "widely" - not sure how we'd measure/justify that.

File: docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+---
+title: "Securing software artifacts with Tekton Chains and IBM's DevSecOps"
+author: "Arnaud J Le Hors"
+is_guest_post: true
+---
+
+Tekton Chains and the IBM DevSecOps offering that builds on it can now be used to secure software artifacts with SLSA.
```
Comment (david-a-wheeler) [inline docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md:7]: This is hard to parse. I read the second part as "the IBM DevSecOps". I suggest adding commas to clarify this:

Tekton Chains, and the IBM DevSecOps offering that builds on it,

File: docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+Obviously, it would make sense to have more straightforward values and for the v1 format to be the default. I will do what I can to help make this happen, but in the meantime you now have the right configuration to get the desired result.
+
+## IBM DevSecOps
+
+The [IBM DevSecOps offering](https://cloud.ibm.com/docs/devsecops) available to IBM Cloud users is based on Tekton and Tekton Chains and since January this year one can use it as a SLSA Level 3 build platform producing SLSA Provenance v1 attestations.
+
+This build platform is used internally to produce software artifacts such as the [IBM Cloud Paks](https://www.ibm.com/cloud-paks) generating along with them both SBOMs and SLSA Provenance attestations, and IBM Cloud users can equally benefit from it by using it to produce their own software artifacts.
```
Comment (david-a-wheeler) [inline docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md:31]: I suggest "used internally" -> "used internally by IBM" for clarity.

File: docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+I'm happy to say that the configuration is a bit more straightforward than that of Tekton Chains and one can simply set the `slsa-attestation` parameter to `1` to get the build platform to produce SLSA attestations in the Provenance v1 format.
```
Comment (david-a-wheeler) [inline docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md:33]: after "that the configuration" add "of IBM DevSecOps".

File: docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+Indeed, one must set the Tekton Chains configuration output format option to `slsa/v2alpha3` to get the Provenance v1 format, while setting the format to `slsa/v1` (which is currently the default) produces an attestation in Provenance v0.2. These values are obviously unfortunate. There is actually evidence in the repo that they weren't meant to be exposed to the user but, as it sometimes happens, things that are meant to be temporary stick beyond the initial phase and, as specified in the [How to configure Tekton Chains](https://tekton.dev/docs/chains/slsa-provenance/#how-to-configure-tekton-chains) documentation, `slsa/v2alpha3` is what you must use for now.
+
+Obviously, it would make sense to have more straightforward values and for the v1 format to be the default. I will do what I can to help make this happen, but in the meantime you now have the right configuration to get the desired result.
```
Comment (david-a-wheeler) [inline docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md:25]: Can you talk more about how you will try to make this happen?

**Comment by david-a-wheeler** [comment]
Overall looks fine, see my comments above.

**Comment by lehors** [comment]
> Overall looks fine, see my comments above.

Thanks @david-a-wheeler for your careful review, I've accepted all your proposed changes.

**Comment by lehors** [comment]
@david-a-wheeler can you please approve so this can be merged? Thank you.

File: docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+## IBM DevSecOps
+
+The [IBM DevSecOps offering](https://cloud.ibm.com/docs/devsecops) available to IBM Cloud users is based on Tekton and Tekton Chains and since January this year one can use it as a SLSA Level 3 build platform producing SLSA Provenance v1 attestations.
+
+This build platform is used internally by IBM to produce software artifacts such as the [IBM Cloud Paks](https://www.ibm.com/cloud-paks) generating along with them both SBOMs and SLSA Provenance attestations, and IBM Cloud users can equally benefit from it by using it to produce their own software artifacts.
+
+I'm happy to say that the configuration of IBM DevSecOps is a bit more straightforward than that of Tekton Chains and one can simply set the `slsa-attestation` parameter to `1` to get the build platform to produce SLSA attestations in the Provenance v1 format.
+
+You can then use [cosign](https://github.com/sigstore/cosign) to verify the signature on the produced attestation.
```
Comment (joshuagl) [inline docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md:35]: What about [verifying the artefacts](https://slsa.dev/spec/v1.0/verifying-artifacts) themselves (through provenance)? Is that done with [`cocao locker evidence check`](https://cloud.ibm.com/docs/devsecops?topic=devsecops-cd-devsecops-cli#locker-evidence-check)?

**Review by joshuagl — APPROVED** [review]
Great post. It would be even better if it mentioned how to [verify artefacts](https://slsa.dev/spec/v1.0/verifying-artifacts) not just the signature on the produced attestation.

File: docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md (hunk 8 above)
Comment (lehors) [reply docs/_posts/2024-04-16-tekton-chains-ibm-devsecops.md:35]: As far as I know, this is currently "left as an exercise to the reader" so to speak. :-) But I agree with you that this should be addressed. I'll merge this now because it's been sitting there for a long time already and I'll post an update ASAP.
Thanks for your review!
